src/controller/python/README.md

```diff
@@ -1,244 +1,255 @@
 
-> Running as root via `sudo` to ensure permissions to interface with the
-> bluetooth adapter.
+### 3. Discover Matter accessory device over Bluetooth LE
+
+An uncommissioned accessory device advertises over Bluetooth LE. Run the following command to scan all advertised Matter devices:
 
 ```
-sudo chip-device-ctrl
+chip-device-ctrl > ble-scan
 ```
 
-or select the bluetooth interface by command line arguments.
+### 4. Connect to Matter accessory device over Bluetooth LE
+
+The controller uses a 12-bit value called **discriminator** to discern between multiple commissionable device advertisements. Moreover, a 27-bit **PIN code** is used by the controller to authenticate in the device. You can find those values in the logging terminal of the device (e.g. UART), as listed below.
 
 ```
-sudo chip-device-ctrl --bluetooth-adapter=hci2
+I: 254 [DL]Device Configuration:
+I: 257 [DL] Serial Number: TEST_SN
+I: 260 [DL] Vendor Id: 9050 (0x235A)
+I: 263 [DL] Product Id: 20043 (0x4E4B)
+I: 267 [DL] Product Revision: 1
+I: 270 [DL] Setup Pin Code: 20202021
+I: 273 [DL] Setup Discriminator: 3840 (0xF00)
+I: 278 [DL] Manufacturing Date: (not set)
+I: 281 [DL] Device Type: 65535 (0xFFFF)
 ```
 
-2. Scan BLE devices
+Run the following command to establish secure connection over Bluetooth LE, with the following assumptions for the Matter accessory device:
+ - The discriminator of the device is 3840
+ - The setup pin code of the device is 20202021
+ - The temporary Node ID is 1234
 
 ```
-chip-device-ctrl > ble-scan
-2021-01-19 02:27:23,653 ChipBLEMgr   INFO     scanning started
-2021-01-19 02:27:25,144 ChipBLEMgr   INFO     Name            = CHIP-1383
-2021-01-19 02:27:25,144 ChipBLEMgr   INFO     ID              = ae0125dc-e621-3e05-9166-70ca7ea07985
-2021-01-19 02:27:25,146 ChipBLEMgr   INFO     RSSI            = -32
-2021-01-19 02:27:25,147 ChipBLEMgr   INFO     Address         = DC:A6:32:A5:4C:56
-2021-01-19 02:27:25,151 ChipBLEMgr   INFO     Pairing State   = 0
-2021-01-19 02:27:25,151 ChipBLEMgr   INFO     Discriminator   = 1383
-2021-01-19 02:27:25,152 ChipBLEMgr   INFO     Vendor Id       = 9050
-2021-01-19 02:27:25,152 ChipBLEMgr   INFO     Product Id      = 65279
-2021-01-19 02:27:25,155 ChipBLEMgr   INFO     Adv UUID        = 0000fff6-0000-1000-8000-00805f9b34fb
-2021-01-19 02:27:25,156 ChipBLEMgr   INFO     Adv Data        = 0067055a23fffe
-2021-01-19 02:27:27,257 ChipBLEMgr   INFO
-2021-01-19 02:27:34,213 ChipBLEMgr   INFO     scanning stopped
-Connect to BLE device
+chip-device-ctrl > connect -ble 3840 20202021 1234
 ```
 
-3.  Set wifi credential
+You may skip providing the last parameter which is a Node ID. In such case, the controller will randomly assign it. However, note the Node ID down, because it is required later in the configuration process.
```

Comment:
```suggestion
You can skip the last parameter, that is the Node ID. If you skip it, the controller will assign it randomly. However, note the Node ID down, because it is required later in the configuration process.
```